BangazonAPI/Controllers/DepartmentController.cs

```diff
@@ -0,0 +1,235 @@
+
+            else if (_include == "employees") //this part should give you the employees as well
+            {
+                sql = @"
+                    SELECT
+                    d.Id,
+                    d.Name,
+                    d.Budget,
+                    e.id,
+                    e.lastName,
+                    e.firstName,
+                    e.DepartmentId,
+                    e.isSupervisor
+                    FROM Department d
+                    JOIN Employee e ON e.DepartmentId = d.Id";
+
+                   
+                
+                using (IDbConnection conn = Connection)
+                {
+                    Dictionary<int, Department> departmentEmployees = new Dictionary<int, Department>();
+
+                    IEnumerable<Department> departments = await conn.QueryAsync<Department, Employee, Department>(
+                    sql,
+                    (department, employee) =>
+                    {
+                        if (!departmentEmployees.ContainsKey(department.Id))
+                        {
+                            departmentEmployees[department.Id] = department;
+                        }
+                        departmentEmployees[department.Id].Employees.Add(employee);
+                        return department;
+                    });
+                    return Ok(departmentEmployees.Values);
+
+                }
+            }
+
+            else  //will give you all departments only
+            {
+                     sql = @"
+                    SELECT
+                    d.Id,
+                    d.Name,
+                    d.Budget
+                    FROM Department d
+                    ";
+                using (IDbConnection conn = Connection)
+                {
+
+                    IEnumerable<Department> departments = await conn.QueryAsync<Department>(
+                        sql);
+
+                    return Ok(departments);
+                }
+            }
+            
+           
+
+           
+        }
+
+        // GET api/departments/5
+        [HttpGet("{id}", Name = "GetDepartment")]
+        public async Task<IActionResult> Get([FromRoute]int id)
+        {
+            string sql = $@"
+            SELECT
+                d.Id,
+                d.Name,
+                d.Budget
+            FROM Department d
+            WHERE d.Id = {id}
+            ";
+
+            using (IDbConnection conn = Connection)
+            {
+                IEnumerable<Department> departments = await conn.QueryAsync<Department>(sql);
+                return Ok(departments.Single());
+            }
+        }
+
+        // POST api/Department
+        [HttpPost]
+        public async Task<IActionResult> Post([FromBody] Department departments)
+        {
+            string sql = $@"INSERT INTO Department
+            (Name, Budget)
+            VALUES
+            (
+                '{departments.Name}',
+                '{departments.Budget}'
```

Comment:
Since `Budget` is a number, you don't need quotes

Reply:
Good to know thanks